gui/launcher.cpp

```diff
@@ -804,9 +662,558 @@ void LauncherDialog::reflowLayout() {
+				color = ThemeEngine::kFontColorAlternate;
+				// If more conditions which grey out entries are added we should consider
+				// enabling this so that it is easy to spot why a certain game entry cannot
+				// be started.
+
+				// description += Common::String::format(" (%s)", _("Not found"));
+			}
+		}
+
+		l.push_back(iter->description);
+		colors.push_back(color);
+		_domains.push_back(iter->key);
+	}
+
+	const int oldSel = _list->getSelected();
+	_list->setList(l, &colors);
+	if (oldSel < (int)l.size())
+		_list->setSelected(oldSel);	// Restore the old selection
+	else if (oldSel != -1)
+		// Select the last entry if the list has been reduced
+		_list->setSelected(_list->getList().size() - 1);
+	updateButtons();
+
+	// Update the filter settings, those are lost when "setList"
+	// is called.
+	_list->setFilter(_searchWidget->getEditString());
+}
+
+void LauncherSimple::handleKeyDown(Common::KeyState state) {
+	if (state.keycode == Common::KEYCODE_TAB) {
+		// Toggle between the game list and the quick search field.
+		if (getFocusWidget() == _searchWidget) {
+			setFocusWidget(_list);
+		} else if (getFocusWidget() == _list) {
+			setFocusWidget(_searchWidget);
+		}
+	}
+	Dialog::handleKeyDown(state);
+	updateButtons();
+}
+
+void LauncherSimple::handleCommand(CommandSender *sender, uint32 cmd, uint32 data) {
+
+	switch (cmd) {
+	case kListItemActivatedCmd:
+	case kListItemDoubleClickedCmd:
+		LauncherDialog::handleCommand(sender, kStartCmd, 0);
+		break;
+	case kListItemRemovalRequestCmd:
+		LauncherDialog::handleCommand(sender, kRemoveGameCmd, 0);
+		break;
+	case kListSelectionChangedCmd:
+		updateButtons();
+		break;
+	case kSearchCmd:
+		// Update the active search filter.
+		_list->setFilter(_searchWidget->getEditString());
+		break;
+	case kSearchClearCmd:
+		// Reset the active search filter, thus showing all games again
+		_searchWidget->setEditString(Common::U32String());
+		_list->setFilter(Common::U32String());
+		break;
+	default:
+		LauncherDialog::handleCommand(sender, cmd, data);
+	}
+}
+
+void LauncherSimple::updateButtons() {
+	bool enable = (_list->getSelected() >= 0);
+	if (enable != _startButton->isEnabled()) {
+		_startButton->setEnabled(enable);
+		_startButton->markAsDirty();
+	}
+	if (enable != _editButton->isEnabled()) {
+		_editButton->setEnabled(enable);
+		_editButton->markAsDirty();
+	}
+	if (enable != _removeButton->isEnabled()) {
+		_removeButton->setEnabled(enable);
+		_removeButton->markAsDirty();
+	}
+
+	int item = _list->getSelected();
+	bool en = enable;
+
+	if (item >= 0)
+		en = !(Common::checkGameGUIOption(GUIO_NOLAUNCHLOAD, ConfMan.get("guioptions", _domains[item])));
+
+	if (en != _loadButton->isEnabled()) {
+		_loadButton->setEnabled(en);
+		_loadButton->markAsDirty();
+	}
+}
+
+#pragma mark -
+
+#ifndef DISABLE_LAUNCHERDISPLAY_GRID
+LauncherGrid::LauncherGrid(const U32String &title)
+	: LauncherDialog(title), 
+	_grid (nullptr), _addButton(nullptr), _startButton(nullptr),
```

Comment:
formatting is not correct.

Reply:
Corrected formatting in commits 739cb8e and 915d669.